widget/button.go

```diff
@@ -358,7 +358,7 @@ func (r *buttonRenderer) applyTheme() {
 	r.button.applyButtonTheme()
 	r.label.Segments[0].(*TextSegment).Style.ColorName = theme.ColorNameForeground
 	switch {
-	case r.button.disabled:
+	case r.button.Disabled():
```

Comment:
If we are going with the idea that public methods are called unlocked but private should be within the lock then this isn't the right fix.
If that mantra isn't going to work going forward then we should make another plan for setting expecation of lock/thread safety as we go more into resolving these issues.

Reply:
That would be more consistent, yes. If it seems ugly there could be an internal function doing that?

Reply:
A little more to write but I wouldn't call it ugly :)